src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketSender.cs

```diff
@@ -68,18 +72,14 @@ private List<ArraySegment<byte>> GetBufferList(in ReadOnlySequence<byte> buffer)
             {
                 _bufferList = new List<ArraySegment<byte>>();
             }
-            else
-            {
-                // Buffers are pooled, so it's OK to root them until the next multi-buffer write.
-                _bufferList.Clear();
```

Comment:
How much memory is expected to be used in the bufferlist?

Reply:
Are you asking how big the list will get?

Reply:
As we aren't clearing here, how much idle memory will be in the bufferlist?

Reply:
We clear in Return.